cawemo-to-web-modeler-migration/migrate-cawemo-to-web-modeler.sh

```diff
@@ -0,0 +1,210 @@
+
+PROJECTS=$(cat .curl_tmp)
+
+if [ "$HTTP_STATUS" != "200" ]
+then
+    echo "⚠ GET https://cawemo.com/api/v1/projects failed with status code $HTTP_STATUS"
+    echo "[$(date +%s)] ⚠ GET https://cawemo.com/api/v1/projects failed with status code $HTTP_STATUS" >> migration.log
+    
+    echo "Aborted"
+    exit 1
+fi
+
+TOKEN=$(curl -s --header "Content-Type: application/json" --request POST --data "{\"grant_type\":\"client_credentials\", \"audience\":\"api.cloud.camunda.io\", \"client_id\":\"$MODELER_CLIENT_ID\", \"client_secret\":\"$MODELER_CLIENT_SECRET\"}" https://login.cloud.camunda.io/oauth/token | jq -r .access_token)
+
+for row in $(echo $PROJECTS | jq -r '.[] | @base64'); do
+    _jq() {
+     echo ${row} | base64 --decode | jq -r ${1}
+    }
+
+    OLD_PROJECT_ID=$(_jq '.id')
+    PROJECT_NAME=$(_jq '.name')
+
+    echo "Migrating project ${GREEN}$PROJECT_NAME${NC} (ID: $OLD_PROJECT_ID)..."
+
+    HTTP_STATUS=$(curl -s -o .curl_tmp -w "%{http_code}" -X POST "https://modeler.cloud.camunda.io/api/beta/projects" \
+        -H "Content-Type: application/json" \
+        -H "Authorization: Bearer $TOKEN" \
+        -d "$(_jq '.')")
+
+    NEW_PROJECT=$(cat .curl_tmp)
+
+    if [ "$HTTP_STATUS" != "200" ]
+    then
+        echo "⚠ POST https://modeler.cloud.camunda.io/api/beta/projects failed with status code $HTTP_STATUS"
+        echo "[$(date +%s)] ⚠ POST https://modeler.cloud.camunda.io/api/beta/projects failed with status code $HTTP_STATUS" >> migration.log
+
+        echo "Aborted"
+        exit 1
+    fi
+
+    echo "[$(date +%s)] Migrated project $PROJECT_NAME (ID: $OLD_PROJECT_ID)" >> migration.log
+
+    NEW_PROJECT_ID=$(jq -r .id <<< "$NEW_PROJECT")
+
+    # maps folder IDs to original IDs to not create them twice + create file in right folder
+    declare -A folderIds
+
+    FILES=$(curl -s -u $CAWEMO_USER_ID:$CAWEMO_API_KEY "https://cawemo.com/api/v1/projects/$OLD_PROJECT_ID/files")
+
+    FILE_COUNT=$(echo $FILES | jq '. | length')
+
+    echo "$FILE_COUNT files to be migrated"
+
+    FILE_INDEX=1
+    for file in $(echo $FILES | jq -r '.[] | @base64'); do
+        _file_jq() {
+            echo ${file} | base64 --decode | jq -r ${1}
+        }
+
+        if [ $FILE_COUNT -eq $FILE_INDEX ]
+        then
+            TREE_SYMBOL="└──"
+        else
+            TREE_SYMBOL="├──"
+        fi
+
+        OLD_FILE_ID=$(_file_jq '.id')
+        FILE_NAME=$(_file_jq '.name')
+        FILE_TYPE=$(_file_jq '.type')
+        FILE_PATH=$(_file_jq '.canonicalPath')
+
+        FILE_TREE_SPACES=""
+
+        # Create Folders
+
+        PARENT_ID="null"
+
+        for folder in $(echo $FILE_PATH | jq -r '.[] | @base64'); do
+            _folder_jq() {
+                echo ${folder} | base64 --decode | jq -r ${1}
+            }
+
+            FOLDER_NAME=$(_folder_jq '.name')
+            OLD_FOLDER_ID=$(_folder_jq '.id')
+
+            # skip this folder if already created
+            if [ -z "${folderIds[$OLD_FOLDER_ID]}" ]
```

Comment:
I got an error in this line:
```
./migrate-cawemo-to-web-modeler.sh: line 118: fc98fdfb-0e75: value too great for base (error token is "0e75")
```

Only happens when I run the script directly with `./migrate-cawemo-to-web-modeler.sh`, not with `zsh migrate-cawemo-to-web-modeler.sh` 🤔.

Reply:
Works now when using `#!/bin/zsh`

@jfriedenstab please verify